Makefile

```diff
@@ -15,8 +15,6 @@ CSI_PROXY_API_GEN = $(BUILD_DIR)/csi-proxy-api-gen
 # overrides the $(CMDS:%=build-%) rule in release-tools/build.make because this project is not compatible with go >1.23
 # TODO: remove this override as part of https://github.com/kubernetes-csi/csi-proxy/issues/361
 build-csi-proxy: check-go-version-go
-	# Checks that the go version is 1.22 or lower
```

Comment:
I added this check so that during the build of CSI Proxy if the environment uses >go1.22 then it'd break.

At the same time go.mod doesn't seem to have any changes, the min go version is still go1.22.

Is this PR enough to say that all the privileged storage operations are no longer going to fail with go1.23+? I think that moving to go1.23+ could be a cleanup item after all the refactors.

Reply:
Sure. The second commit was discarded.